.github/workflows/reusable_build_sample_apps.yml

```diff
@@ -74,13 +100,32 @@ jobs:
       - name: Dump GitHub Action metadata because Fastlane uses it. Viewing it here helps debug JSON parsing code in Firebase.
         run: cat $GITHUB_EVENT_PATH
 
+      - name: Prepare Fastlane Build Arguments
+        run: |
+          if [ "${{ inputs.use_latest_sdk_version }}" = "true" ]; then
+            sdk_version="\"sdk_version\":\"${{ steps.latest-sdk-version-step.outputs.LATEST_TAG }}\""
+          else
+            sdk_version=""
+          fi
+          
+          distribution_groups="\"distribution_groups\":\"${{ env.firebase_distribution_groups }}\""
+          
+          # Construct the fastlane-build-args JSON
+          if [ -n "$sdk_version" ]; then
+            fastlane_build_args="{${sdk_version},${distribution_groups}}"
+          else
+            fastlane_build_args="{${distribution_groups}}"
+          fi
+          
+          echo "fastlane_build_args=$fastlane_build_args" >> $GITHUB_ENV
+
       - name: Deploy build via Fastlane
         if: ${{ ! (inputs.use_latest_sdk_version == true && matrix.sample-app == 'kotlin_compose') }}
         uses: maierj/fastlane-action@v3.1.0
         with:
           lane: ${{ inputs.use_latest_sdk_version == true && 'android build_sample_app_for_sdk_release' || 'android build' }}
           subdirectory: "samples/${{ matrix.sample-app }}"
-          options: ${{ inputs.use_latest_sdk_version == true && format('{{"sdk_version":"{0}"}}', steps.latest-sdk-version-step.outputs.LATEST_TAG) || '' }}
```

Comment:
I wanted to avoid breaking current functionality, but keeping this in a single line was getting complicated. So I moved it to the step above for better readability and ease.